docs/src/reviewing_guide.md

```diff
@@ -0,0 +1,98 @@
+7. When requesting data from another node, there are a few things to look out for:
+   1. What happens if the node does not respond? Is there a timeout? The timeout should be set as low as possible, otherwise 
+the node can hold up the processing on the requesting node.
+   2. What happens if the node returns unexpected data?
+8. In cases where we have received data from a node, whether we requested it or not, if it is bad, is the node banned?
+
+
+### More specific things to look out for
+
+#### Usize
+`usize` is a platform dependent type. It is 32 bits on 32 bit platforms and 64 bits on 64 bit platforms. This can cause 
+differences in hashing and serialisation. It should be avoided.
+
+#### Atomics
+It is fine to use `AtomicBool`, `AtomicUsize` and the other atomic types, but all `Orderings` must use `Ordering::SeqCst`.
+See [nomicon](https://doc.rust-lang.org/nomicon/atomics.html) for more detail, but most of the ordering enum values have little visible effect on intel based architectures([x86](https://simple.wikipedia.org/wiki/X86) and [x86-64](https://en.wikipedia.org/wiki/X86-64)
+and are only seen in arm based systems. It’s also unlikely that any performance is gained from using a different ordering, so rather be on the safe side and use `SeqCst` everywhere.
+
+It is also pretty much impossible to test ordering on intel based systems, so it is best to avoid it.
+
+#### Vec
+When using `Vec::with_capacity`, is the size input provided by an untrusted party? If so, use a maximum bound on the capacity and the number of items returned or reject the message. The latter is almost always more appropriate. 
+
+#### Unchecked arithmetic and overflows
+Any arithmetic operation has a chance of overflowing or underflowing. It is best to use the checked versions, for example `checked_mul`, `checked_sub` etc. to avoid this.
+Be careful of indexing into a slice or array with an untrusted value. This can cause a panic. Use `get` instead.
+
+#### Shifts (<<, >>)
+For some reason, `checked_shr` and `checked_shl` do not act like `checked_add` and `checked_sub` for overflows. They will only return `None` if the inputted shift is too large, but will 
+shift even if the MSB is set. i.e. `0b1000_0000 << 1` will return `0b0000_0000`. This is often not what we expect. Use `leading_zeroes()` to check if the value would overflow before shifting.
+
+#### Tokio
+Tokio has many useful channels, and it can be difficult to know what to look out for, so here are some things to keep in mind:
+1. When using `watch`:
+   1. Watches can block if the reference returned from `borrow()` is held for a long time. Any call to `borrow()` should drop the reference as soon as possible.
+2. When using `broadcast`:
+   1. If one receiver is not receiving the values, it will return a `Lagged` error. This should be logged, but in most cases the code can continue as normal.
```

Comment:
```suggestion
   1. If one receiver is not receiving the values, it will return a `Lagged` error. This should be logged, but in most cases the code can continue as normal.
   2. When receiving events in a loop, Error::Closed should be used to break out of the loop, because the sender halves have dropped and no more events will be received
```